src/components/ContentNode/TabNavigator.vue

```diff
@@ -71,6 +71,22 @@ export default {
       currentTitle: this.titles[0],
     };
   },
+  watch: {
+    titles(newVal, oldVal) {
+      if (newVal.length < oldVal.length) {
+        // set to the first tab if selected tab was removed
+        const removedTab = oldVal.find(tab => !newVal.includes(tab));
```

Comment:
The `.includes` part implies that there is a nested loop, which isn't necessary. To do the same check without the nested loop, I think you could do something like this:

```javascript
const removedTab = oldVal.find((tab, i) => newVal[i] !== tab));
```